patches/api/0426-Add-Material-getPlacedType-API.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Aya <31237389+tal5@users.noreply.github.com>
+Date: Tue, 15 Aug 2023 10:05:01 +0100
+Subject: [PATCH] Add Material#getPlacedType API
+
+
+diff --git a/src/main/java/org/bukkit/Material.java b/src/main/java/org/bukkit/Material.java
+index 03b47012447430a350e152920f754d993d4023db..ba087077f948bcdfa18708dc1804badb9cebc6b0 100644
+--- a/src/main/java/org/bukkit/Material.java
++++ b/src/main/java/org/bukkit/Material.java
+@@ -11047,4 +11047,23 @@ public enum Material implements Keyed, Translatable, net.kyori.adventure.transla
+     public boolean isEnabledByFeature(@NotNull World world) {
+         return Bukkit.getDataPackManager().isEnabledByFeature(this, world);
+     }
++
++    // Paper start
++    /**
++     * Gets the material that would be placed down by this item.
++     * For most placeable items this is the same type as the item, but it is different for some.
++     * <p>
++     * Some examples include:
++     * <pre>
++     *     {@link #REDSTONE} -> {@link #REDSTONE_WIRE}
++     *     {@link #PUMPKIN_SEEDS} -> {@link #PUMPKIN_STEM}
++     * </pre>
++     * @return The material that would be placed down by this item, or {@code null} if it can't be placed.
++     * @throws IllegalArgumentException if {@link #isItem()} is {@code false}
++     */
++    @Nullable
++    public Material getPlacedType() {
++        return Bukkit.getUnsafe().getPlacedType(this);
++    }
++    // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
+index de36dab37d3af249cf51573d163731c12346066a..b3d524e9f366b254d89d901fe99c827825f0ea9c 100644
+--- a/src/main/java/org/bukkit/UnsafeValues.java
++++ b/src/main/java/org/bukkit/UnsafeValues.java
+@@ -252,4 +252,22 @@ public interface UnsafeValues {
+ 
+     String getStatisticCriteriaKey(@NotNull org.bukkit.Statistic statistic);
+     // Paper end
++
++    // Paper start
++    /**
++     * Gets the material that would be placed down by {@code item}.
++     * For most placeable items this is the same type as the item, but it is different for some.
++     * <p>
++     * Some examples include:
++     * <pre>
++     *     {@link Material#REDSTONE} -> {@link Material#REDSTONE_WIRE}
++     *     {@link Material#PUMPKIN_SEEDS} -> {@link Material#PUMPKIN_STEM}
++     * </pre>
++     * @param item An item material.
++     * @return The material that would be placed down by {@code item}, or {@code null} if it can't be placed.
++     * @throws IllegalArgumentException if {@code item} isn't an item ({@link Material#isItem()} is {@code false})
++     */
```

Comment:
Can drop the javadoc here, just increases workload to maintain it since its a copy